actors/runtime/src/main/java/cloud/orbit/actors/cloner/KryoCloner.java

```diff
@@ -66,49 +73,21 @@ public Kryo create()
             {
                 Kryo kryo = new Kryo();
 
-                // Adding support to clone unmodifiable collections as modifiable
-                // collections.
-
-                // It has to be noted that this might not be the expected behavior
-                // in some cases.
-
-                // Without this the unmodifiable collections cause kryo to throw an
-                // exception.. Class cannot be created (missing no-arg constructor).
-
-                kryo.addDefaultSerializer(Collection.class, new CollectionSerializer()
```

Comment:
Replaced with the custom serializers below. Instead of a linked x (using double~ the memory!) we will end up with a cloned unmodifiable x :)